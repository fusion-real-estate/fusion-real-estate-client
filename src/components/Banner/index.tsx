import * as S from './styles'

export type BannerProps = {
  img: string
  title: string
  subtitle: string
}

const Banner = ({ img, title, subtitle }: BannerProps) => (
  <S.Wrapper>
    <S.Image src={img} role="img" aria-label={title} />

    <S.Caption>
      <S.Title>{title}</S.Title>
      <S.Subtitle dangerouslySetInnerHTML={{ __html: subtitle }} />
    </S.Caption>
  </S.Wrapper>
)

export default Banner
