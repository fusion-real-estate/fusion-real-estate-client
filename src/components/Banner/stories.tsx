import { Story, Meta } from '@storybook/react/types-6-0'
import Banner, { BannerProps } from '.'

export default {
  title: 'Banner',
  component: Banner,
  args: {
    img: 'https://creativelayers.net/themes/houzing-html/images/home/1.jpg',
    title: 'Ã‰ muito bom estar em casa!',
    subtitle: '<h3>Encontre sua casa perfeita</h3>'
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: Story<BannerProps> = (args) => <Banner {...args} />
