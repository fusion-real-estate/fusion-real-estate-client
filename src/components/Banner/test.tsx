import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Banner from '.'

const props = {
  img:
    'https://images.unsplash.com/photo-1570129477492-45c003edd2be?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80',
  title: 'É muito bom estar em casa!',
  subtitle: 'Encontre sua casa perfeita'
}

describe('<Banner />', () => {
  it('should render the heading', () => {
    renderWithTheme(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /é muito bom estar em casa!/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /encontre sua casa perfeita/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: /é muito bom estar em casa!/i })
    ).toBeInTheDocument()
  })
})
