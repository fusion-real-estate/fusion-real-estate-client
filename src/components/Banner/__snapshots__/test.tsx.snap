// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render the heading 1`] = `
.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(https://images.unsplash.com/photo-1570129477492-45c003edd2be?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 1.6rem;
}

.c4 strong {
  color: #0D263B;
  font-weight: 600;
}

@media (max-width:1170px) {
  .c0 .sc-bdvvaa {
    right: 0;
  }

  .c0 .sc-bdvvaa::before {
    display: none;
  }
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width:768px) {
  .c1 {
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="É muito bom estar em casa!"
    class="c1"
    role="img"
    src="https://images.unsplash.com/photo-1570129477492-45c003edd2be?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80"
  />
  <div
    class="c2"
  >
    <h1
      class="c3"
    >
      É muito bom estar em casa!
    </h1>
    <h3
      class="c4"
    >
      Encontre sua casa perfeita
    </h3>
  </div>
</main>
`;
